src/Model/CompanyResponse.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hyra\UkCompaniesHouseLookup\Model;
+
+use Symfony\Component\Serializer\Annotation\SerializedName;
+use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+final class CompanyResponse extends AbstractResponse
+{
+    #[SerializedName('company_name')]
+    #[NotBlank]
+    public string $companyName;
+
+    #[SerializedName('company_number')]
+    #[NotBlank]
+    public string $companyNumber;
+
+    #[NotBlank]
+    public string $type;
+
+    #[SerializedName('company_status')]
+    #[NotBlank]
+    public string $status;
+
+    #[SerializedName('company_status_detail')]
+    public ?string $statusDetail = null;
+
+    #[SerializedName('date_of_creation')]
+    #[NotBlank]
+    public \DateTimeImmutable $dateOfCreation;
+
+    #[NotBlank]
+    public string $jurisdiction;
+
+    #[SerializedName('registered_office_address')]
+    public ?AddressResponse $address = null;
+
+    /**
+     * @var PreviousNamesResponse[]
+     *
+     * @Assert\All({
+     *
+     *     @Assert\Type("Hyra\UkCompaniesHouseLookup\Model\PreviousNamesResponse")
+     * })
+     */
```

Comment:
We can use nested attributes to do this.
```suggestion
    /**
     * @var PreviousNamesResponse[]
     */
     #[Assert\All(constraints: new Assert\Type(PreviousNamesResponse::class))
```

Reply:
@ndench Only if we drop support for PHP 8.0 🙊